servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -109,9 +109,11 @@ public void setWorkerId() throws IOException {
   public void registerWithMaster() {
     BlockStoreMeta storeMeta = mBlockDataManager.getStoreMeta();
     try {
-      mWorkerId =
-          mMasterClient.register(mWorkerId, storeMeta.getCapacityBytesOnTiers(),
-              storeMeta.getUsedBytesOnTiers(), storeMeta.getBlockList());
+      // Request a new worker ID.
+      setWorkerId();
```

Comment:
I don't think registerWithMaster() should set the workerId. Instead, I think handleMasterCommand() should set the worker id before calling register. This is to mirror what happens in BlockWorker.java, since it sets the worker id, then calls register.